core/eolearn/core/eodata.py

```diff
@@ -1030,6 +1080,38 @@ def _parse_feature_value(self, value):
         return value
 
 
+class _Loader:
+    def __init__(self, filesystem, ftype, fname, path):
+        self.filesystem = filesystem
+        self.ftype = ftype
+        self.fname = fname
+        self.path = path
+
+    def load(self):
+        file_handle = BytesIO()
+        self.filesystem.download(self.path, file_handle)
+        file_handle.seek(0)
+
+        if self.path.endswith('.gz'):
+            with gzip.open(file_handle, 'rb') as gzip_fp:
+                return self._decode(gzip_fp, self.path)
+
+        return self._decode(file_handle, self.path)
+
+    @staticmethod
+    def _decode(file, path):
+        if '.pkl' in path:
```

Comment:
To avoid using strings you could do `FileFormat.PICKLE.extension() in path` and similarly everywhere else.

Also, it might be nice if we turn `FileFormat.extension` method into a property.

Reply:
Done.